From 07ac7fad0451cc536a4dbb7682465d2a75049f22 Mon Sep 17 00:00:00 2001
From: NelloKudo <marshnelloosu@gmail.com>
Date: Mon, 14 Sep 2026 17:31:28 +0200
Subject: [PATCH 17/17] makedep: Link unix libraries with the C++ compiler when
 they have C++ sources.

This should be what remains of Proton's "makedep: Allow building modules with
C++ sources." (4aa55a3ae2e) on top of upstream C++ support
---
 tools/makedep.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/tools/makedep.c b/tools/makedep.c
index bababcf36ae..5e847ef15b5 100644
--- a/tools/makedep.c
+++ b/tools/makedep.c
@@ -4082,6 +4082,8 @@ static void output_unix_lib( struct makefile *make )
     struct strarray unix_deps = empty_strarray;
     struct strarray unix_libs = add_unix_libraries( make, &unix_deps );
     unsigned int arch = 0;  /* unix libs are always native */
+    struct incl_file *source;
+    bool has_cxx = false;
     const char *unixlib;
 
     if (make->disabled[arch]) return;
@@ -4093,7 +4095,9 @@ static void output_unix_lib( struct makefile *make )
     output_filenames_obj_dir( make, make->unixobj_files );
     output_filenames( unix_deps );
     output( "\n" );
-    output( "\t%s$(CC) -o $@", cmd_prefix( "CCLD" ));
+    LIST_FOR_EACH_ENTRY( source, &make->sources, struct incl_file, entry )
+        if ((source->file->flags & (FLAG_C_UNIX | FLAG_C_CXX)) == (FLAG_C_UNIX | FLAG_C_CXX)) has_cxx = true;
+    output( "\t%s%s -o $@", cmd_prefix( "CCLD" ), has_cxx ? "$(CXX)" : "$(CC)" );
     output_filenames( get_expanded_make_var_array( make, "UNIXLDFLAGS" ));
     output_filenames_obj_dir( make, make->unixobj_files );
     output_filenames( unix_libs );
-- 
2.55.0

